### DISCUSSION TITLE
Improve build system and support ESM

### DISCUSSION THREAD

**Discussion by winston0410** [medusajs/medusa#4235] [Feature Requests]
Right now medusa is only supporting CJS module, and used babel as the compile/build tool. This setup is 4~5 years older than the modern system, and as Vite and ESM is becoming mainstream now, I think it will be great to modernize the build tool

Two limitations with the current setup:
- cannot use `medusa develop` in a monorepo that installs packages globally(e.g Nx), as it is using a Babel with a hardcoded path
- cannot use ESM package without hack like [fix-esm](https://www.npmjs.com/package/fix-esm)

**Comment by Bernix01** [comment]
I would like to add that npm packages started to drop CJS support rendering them incompatible with medusa.

**Comment by ghardin1314** [comment]
Our company is very on the fence about using Medusa and this could be a big deal breaker. Seems like the underlying tech setup is ~5-8 years behind in terms of bundling, etc. Would like to hear more from the devs on their plans forward with this

**Comment by carstenblt** [comment]
The longer this is unresolved, the more it will become a problem

**Comment by hadronomy** [comment]
Any news on this?
